frontend/main.js

```diff
@@ -405,9 +485,31 @@ async function startApp () {
       // to ensure that we don't override user interactions that have already
       // happened (an example where things can happen this quickly is in the
       // tests).
-      sbp('gi.db/settings/load', SETTING_CURRENT_USER).then(identityContractID => {
-        if (!identityContractID || this.ephemeral.finishedLogin === 'yes') return
-        return sbp('gi.actions/identity/login', { identityContractID }).catch((e) => {
+      sbp('gi.db/settings/load', SETTING_CURRENT_USER).then(async (identityContractID) => {
+        // This loads CHELONIA_STATE when _not_ running as a service worker
+        const cheloniaState = await sbp('gi.db/settings/load', 'CHELONIA_STATE')
+        if (!cheloniaState || !identityContractID) return
+        if (cheloniaState.loggedIn?.identityContractID !== identityContractID) return
+        const contractSyncPriorityList = [
+          'gi.contracts/identity',
+          'gi.contracts/group',
+          'gi.contracts/chatroom'
+        ]
+        const getContractSyncPriority = (key) => {
+          const index = contractSyncPriorityList.indexOf(key)
+          return index === -1 ? contractSyncPriorityList.length : index
+        }
+        await sbp('chelonia/contract/sync', identityContractID, { force: true })
+        const contractIDs = Object.keys(cheloniaState.contracts)
+        await Promise.all(Object.entries(contractIDs).sort(([a], [b]) => {
+          // Sync contracts in order based on type
+          return getContractSyncPriority(a) - getContractSyncPriority(b)
+        }).map(([, ids]) => {
+          return sbp('okTurtles.eventQueue/queueEvent', `appStart:${identityContractID ?? '(null)'}`, ['chelonia/contract/sync', ids, { force: true }])
+        }))
```

Comment:
All of this code appears to be a DRY violation from duplication in `'gi.actions/identity/login'`, is it possible to consolidate it to one place?